extensions-utils/src/dfx.rs

```diff
@@ -50,6 +54,57 @@ where
         })
 }
 
+/// Calls a binary that was delivered with an extension tarball.
+///
+/// # Returns
+/// - On success, returns stdout as a string.
+/// - On error, returns an error message including stdout and stderr.
+#[context("Calling {} CLI failed, or, it returned an error.", binary_name)]
+pub fn call_extension_bundled_binary<S, I>(binary_name: &str, args: I) -> anyhow::Result<String>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let extension_binary_path =
+        std::env::current_exe().map_err(|e| anyhow::anyhow!("Failed to get current exe: {}", e))?;
+    let extension_dir_path = extension_binary_path
+        .parent()
+        .unwrap_or_else(|| Path::new("."));
+    let binary_to_call = extension_dir_path.join(binary_name);
+
+    let mut command = Command::new(&binary_to_call);
+    command.args(args);
+    // The sns command line tool itself calls dfx; it should call this dfx.
+    // The sns command line tool should not rely on commands not packaged with dfx.
+    // The same applies to other bundled binaries.
+    command.env(
+        "PATH",
+        binary_to_call.parent().unwrap_or_else(|| Path::new(".")),
```

Comment:
> is binary_to_call.parent() the same as extension_dir_path?

yes, thank you

> where is the path to dfx coming from?

Why would it need a path to dfx? I think we're not concerned with `dfx` at this point because we've already localized the extension:
1. user calls `dfx EXT`
2. `dfx` version n.n.n could redirect execution to `dfx` version o.o.o (if we're in dir with dfx.json that specifies dfx version)
3. `dfx` executes extension BIN in 
   - `.cache/dfinity/versions/o.o.o/extensions/EXT/BIN` , or
   - `.cache/dfinity/versions/n.n.n/extensions/EXT/BIN` 
4. `BIN` does not care if it's in `o.o.o/extensions/EXT` or `n.n.n/extensions/EXT`. It only cares if it can find binaries in the same dir as `BIN` is in (aka `std::env::current_exe()`)